packages/core/src/asset/ResourceManager.ts

```diff
@@ -176,6 +176,26 @@ export class ResourceManager {
     this._contentRestorerPool[restorer.resource.instanceId] = restorer;
   }
 
+  /**
+   * Callback sub asset on success.
+   * @param assetURL - The url of sub asset
+   * @param value - Sub asset
+   */
+  onSubAssetSuccess<T>(assetURL: string, value: T): void {
+    this._pendingPromise[assetURL]?.resolve(value);
+    delete this._pendingPromise[assetURL];
+  }
```

Comment:
??? You shouldn't add any API in this PR

Reply:
done